pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -282,6 +294,121 @@ func configMapDeleted(obj interface{}) {
 	}
 }
 
+// initVolumeHandleToPvcMap performs all the operations required to initialize the volume id to PVC name map.
+// It also watches for PV update & delete operations, and updates the map accordingly.
+func initVolumeHandleToPvcMap(ctx context.Context) {
+	log := logger.GetLogger(ctx)
+	log.Debugf("Initializing volume ID to PVC name map")
+	k8sOrchestratorInstance.volumeIDToPvcMap = make(map[string]string)
+
+	// Set up kubernetes resource listener to listen events on PersistentVolumes and PersistentVolumeClaims
+	k8sOrchestratorInstance.informerManager.AddPVListener(
+		func(obj interface{}) { // Add
+			pvAdded(obj)
+		},
+		func(oldObj interface{}, newObj interface{}) { // Update
+			pvUpdated(oldObj, newObj)
+		},
+		func(obj interface{}) { // Delete
+			pvDeleted(obj)
+		})
+
+	k8sOrchestratorInstance.informerManager.AddPVCListener(
+		func(obj interface{}) { // Add
+			pvcAdded(obj)
+		},
+		nil, // Update
+		nil, // Delete
+	)
+}
+
+// Since informerManager's sharedInformerFactory is started with no resync period, it never syncs the
+// existing cluster objects to its Store when it's started.
+// pvcAdded provides no additional handling but it ensures that existing PVCs in the cluster
+// gets added to sharedInformerFactory's Store before it's started. Then using informerManager's
+// PVCLister should find the existing PVCs as well.
+func pvcAdded(obj interface{}) {}
+
+// pvAdded adds a volume to the volumeIDToPvcMap if it's already in Bound phase.
+// This ensures that all existing PVs in the cluster are added to the map, even across container restarts.
+func pvAdded(obj interface{}) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	ctx = logger.NewContextWithLogger(ctx)
+	log := logger.GetLogger(ctx)
+
+	pv, ok := obj.(*v1.PersistentVolume)
+	if pv == nil || !ok {
+		log.Warnf("pvAdded: unrecognized object %+v", obj)
+		return
+	}
+
+	if pv.Spec.CSI != nil && pv.Spec.CSI.Driver == csitypes.Name &&
+		pv.Spec.ClaimRef != nil && pv.Status.Phase == v1.VolumeBound {
```

Comment:
We should not be caching file volumes. Shouldn't you also check that this is a ReadWriteOnce PV? Same comment in the other event handlers.

Reply:
Done.